docs/tutorials/graphql.md

```diff
@@ -0,0 +1,257 @@
+      - github.com/99designs/gqlgen/graphql.Int32
+```
+
+## 3. Create Encore service
+
+Now it's time to create our Encore service that will provide the GraphQL API.
+
+First generate the gqlgen boilerplate:
+
+```shell
+$ mkdir -p graphql/generated graphql/model
+$ echo "package model" > graphql/model/model.go
+$ go run github.com/99designs/gqlgen generate
+```
+
+This will create a bunch of files in the `graphql` directory.
+
+Next, create a `graphql/service.go` file containing:
+
+```go
+-- graphql/service.go --
+package graphql
+
+import (
+    "net/http"
+
+    "encore.app/graphql/generated"
+    "github.com/99designs/gqlgen/graphql/handler"
+    "github.com/99designs/gqlgen/graphql/playground"
+)
+
+//go:generate go run github.com/99designs/gqlgen generate
+
+//encore:service
+type Service struct {
+    srv        *handler.Server
+    playground http.Handler
+}
+
+func initService() (*Service, error) {
+    srv := handler.NewDefaultServer(generated.NewExecutableSchema(generated.Config{Resolvers: &Resolver{}}))
+    pg := playground.Handler("GraphQL Playground", "/graphql")
+    return &Service{srv: srv, playground: pg}, nil
+}
+
+//encore:api public raw path=/graphql
+func (s *Service) Query(w http.ResponseWriter, req *http.Request) {
+    s.srv.ServeHTTP(w, req)
+}
+
+//encore:api private raw path=/graphql/playground
+func (s *Service) Playground(w http.ResponseWriter, req *http.Request) {
+    s.playground.ServeHTTP(w, req)
+}
+```
+
+This creates an Encore service that exposes the `/graphql` and `/graphql/playground` endpoints.
+The playground is a private route so it can be accessed from the browser when developing locally, but not in production.
+
+It also adds a `//go:generate` directive that lets you re-run the gqlgen code generation
+by running `go generate ./graphql`.
+
+## 4. Add GraphQL schema
+
+Now it's time to define the GraphQL schema. Create a `graphql/schema.graphqls` file containing:
+
+```
+-- graphql/url.graphqls --
+type Query {
+  urls: [URL!]!
+  get(id: ID!): URL!
+}
+
+type Mutation {
+  shorten(input: String!): URL!
+}
+
+type URL {
+  id:  ID!     # shortened id
+  url: String! # full URL
+}
+```
+
+Then, re-run the code generation to generate the resolver stubs:
+
+```shell
+$ go generate ./graphql
+```
+The stubs will be written to `graphql/url.resolvers.go` and will contain a bunch of unimplemented resolver methods
+that look something like this:
+
+```go
+// Shorten is the resolver for the shorten field.
+func (r *mutationResolver) Shorten(ctx context.Context, input string) (*url.URL, error) {
+	panic(fmt.Errorf("not implemented: Shorten - shorten"))
+}
+```
+
+## 5. Implement resolvers
+
+Now, modify the resolvers to call the `url` service. Since the GraphQL API uses the same types
+(thanks to the `autobind` directive in `gqlgen.yml`)  as the Encore API exposes we can just call the
+endpoints directly. Implement the resolvers in `graphql/url.resolvers.go` like this:
```

Comment:
```suggestion
Now, modify the resolvers to call the `url` service. Since the GraphQL API uses the same types
(thanks to the `autobind` directive in `gqlgen.yml`) as the Encore API exposes, we can just call the
endpoints directly. Implement the resolvers in `graphql/url.resolvers.go` like this:
```